panther_diagnostics/test/test_system_status.cpp

```diff
@@ -70,117 +64,37 @@ SystemStatusTest::SystemStatusTest()
-    panther_diagnostics::SystemStatus::kTemperatureInfoFilename)));
-}
-
 TEST_F(SystemStatusTest, CheckTemperatureReadings)
 {
   const std::string temperature_file_name = testing::TempDir() + "panther_diagnostics_temperature";
   std::filesystem::remove(temperature_file_name);
   EXPECT_FALSE(std::filesystem::exists(temperature_file_name));
 
-  std::ofstream temperature_file(temperature_file_name);
-  temperature_file << "36600";
+  std::ofstream temperature_file(temperature_file_name, std::ofstream::out);
+  temperature_file << 36600 << std::endl;
   temperature_file.close();
 
-  auto temperature = system_status_->GetCPUTemperature(temperature_file_name);
+  const auto temperature = system_status_->GetCoreTemperature(temperature_file_name);
   EXPECT_FLOAT_EQ(temperature, 36.6);
   std::filesystem::remove(temperature_file_name);
 }
 
 TEST_F(SystemStatusTest, CheckMemoryReadings)
 {
-  const std::string memory_file_name = testing::TempDir() + "panther_diagnostics_memory";
-  std::filesystem::remove(memory_file_name);
-  EXPECT_FALSE(std::filesystem::exists(memory_file_name));
-
-  std::string content = R"(
-    MemTotal:        4000000 kB
-    MemFree:         1000000 kB
-    MemAvailable:    2000000 kB
-    Buffers:               0 kB
-    Cached:                0 kB
-  )";
-
-  std::ofstream memory_file(memory_file_name);
-  memory_file << content;
-  memory_file.close();
-  auto memory = system_status_->GetMemoryUsage(memory_file_name);
-  EXPECT_FLOAT_EQ(memory, 50.0);
-  std::filesystem::remove(memory_file_name);
+  const auto memory = system_status_->GetMemoryUsage();
+  EXPECT_TRUE((memory >= 0.0) && (memory <= 100.0));
 }
 
 TEST_F(SystemStatusTest, CheckCPUReadings)
 {
-  const std::string cpu_file_name = testing::TempDir() + "panther_diagnostics_cpu";
-  std::filesystem::remove(cpu_file_name);
-  EXPECT_FALSE(std::filesystem::exists(cpu_file_name));
-
-  const std::string first_context = R"(
-    1000 1000 1000 5000 1000 0 1000 0 0 0
-  )";
-
-  CreateCPUSageFile(cpu_file_name, first_context);
-
-  auto usages = system_status_->GetCPUsUsages(cpu_file_name);
-  for (const auto & usage : usages) {
-    EXPECT_FLOAT_EQ(usage, 0.0);
-  }
-  EXPECT_FLOAT_EQ(system_status_->GetCPUMeanUsage(), 0.0);
-  std::filesystem::remove(cpu_file_name);
-
-  const std::string second_context = R"(
-    1000 500 500 4500 500 0 1000 0 0 0
-  )";
-
-  CreateCPUSageFile(cpu_file_name, second_context);
-  usages = system_status_->GetCPUsUsages(cpu_file_name);
+  const auto usages = system_status_->GetCoresUsages();
 
   for (const auto & usage : usages) {
-    std::cout << usage << std::endl;
-    EXPECT_FLOAT_EQ(usage, 75.0);
+    EXPECT_TRUE((usage >= 0.0) && (usage <= 100.0));
   }
 
-  EXPECT_FLOAT_EQ(system_status_->GetCPUMeanUsage(), 75.0);
-  std::filesystem::remove(cpu_file_name);
+  const auto mean = system_status_->GetCoreMeanUsage(usages);
+  EXPECT_TRUE((mean >= 0.0) && (mean <= 100.0));
 }
```

Comment:
I propose to split this unit case into two separate ones:
- AAA rule
- we test **units**

Reply:
The idea was to create two separate unit tests:
- first to test `GetCoresUsages()` method
- second to test `GetCPUMeanUsage()` (I propose to feed this method with predefined values, so we can verify the exact value of calculated mean usage and not relay on output of the previous method)